presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionModule.java

```diff
@@ -44,6 +44,11 @@ public void configure(Binder binder)
                     config.setMaxConnectionsPerServer(250);
                 });
         binder.bind(PrestoSparkWorkerProperty.class).in(Scopes.SINGLETON);
-        binder.bind(NativeExecutionProcessFactory.class).in(Scopes.SINGLETON);
+        if (System.getProperty("NATIVE_PORT") != null) {
```

Comment:
should we remove the redundant check at https://github.com/prestodb/presto/blob/master/presto-native-execution/src/test/java/com/facebook/presto/spark/AbstractTestPrestoSparkQueries.java#L137 ?

Reply:
Good catch @miaoever . Looks like there are a few things we can clean up as a part of this. Will update

Reply:
Not sure what line it is as that class is no longer there. But things have changed recently so we should double check. Thanks